### DISCUSSION TITLE
All wordpress media items is not showing in graphql-ide

### DISCUSSION THREAD

**Discussion by sanky72** [wp-graphql/wp-graphql#2240] [Q&A] [answered]
Hi, 
I have 23 images in wordpress media library.
I am using gatsby-source-wordpress.
But gatsby's graphql ide only shows 9 media item on this **static** query : 
```
query {
    allWpMediaItem(limit: 30) {
      edges {
        node {
          sourceUrl
          localFile {
            publicURL
            childImageSharp {
              fluid(maxWidth: 800) {
                ...GatsbyImageSharpFluid
              }
            }
          }
        }
      }
    }
  }
```

but I can see that in wordpress's wp-graphql's ide - I am getting all the image when I run this query 
```
query MyQuery {
  mediaItems(first: 25) {
    edges {
      node {
        id
        title
      }
    }
  }
}
```

Please if there's any information which I am missing.

cc @jasonbahl

**Comment by jasonbahl** [comment, marked as answer]
@sanky72 are the images in the media library used within content of pages/posts/custom post types?

Gatsby tries to only load media that is used in the site content instead of loading the entire Media Library. Many sites on WordPress have a lot of images uploaded that they never use on the site, and it would be a lot of extra downloading if Gatsby downloaded every possible image. 

I believe there are options in the Gatsby Source WordPress plugin to change this behavior: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-wordpress/docs/plugin-options.md#typemediaitem

**Reply by sanky72** [reply]
Hey @jasonbahl 
Thanks for replying.
Yes, every image is getting used in one post or another.
I also looked into what media exactly gatsby is loading, found out that they were only featured images, inline images which were used inside posts aren't loaded by gatsby.

I will  surely look at the options of gatsby-source-wordpress for further info.
